cosmos-tx/src/lib.rs

```diff
@@ -1,20 +1,35 @@
 //! Transaction builder and signer for Cosmos-based blockchains
 
+#![cfg_attr(docsrs, feature(doc_cfg))]
 #![doc(
     html_logo_url = "https://raw.githubusercontent.com/cosmos/cosmos-rust/main/.images/cosmos.png",
     html_root_url = "https://docs.rs/cosmos-sdk-proto/0.2.0-pre"
 )]
 #![forbid(unsafe_code)]
 #![warn(trivial_casts, trivial_numeric_casts, unused_import_braces)]
 
+pub mod bank;
+pub mod tx;
+
+mod base;
 mod builder;
 mod decimal;
 mod error;
-mod msg;
+mod prost_ext;
+mod public_key;
 mod signing_key;
 
 pub use crate::{
-    builder::Builder, decimal::Decimal, error::Error, msg::Msg, signing_key::SigningKey,
+    base::{AccountId, Coin, Denom},
+    builder::Builder,
+    decimal::Decimal,
+    error::{Error, Result},
+    public_key::PublicKey,
+    signing_key::SigningKey,
 };
-pub use k256::ecdsa::{Signature, VerifyingKey};
-pub use tendermint::PublicKey;
+
+pub use k256::ecdsa::Signature;
+pub use tendermint;
+
+#[cfg(feature = "rpc")]
```

Comment:
I'm on the fence about the usefulness of an RPC feature, since we're pulling in Tonic as a transitive dep from cosmos-sdk-proto the only advantage an RPC feature gives is keeping us from using any tonic calls and then hoping LTO removes tonic from the final binary.

We should be able to get the same results by just not using RPC calls in a final binary, unless I'm missing somthing.

Thoughts?

Reply:
We need it for the tests anyway, so it can either exist as test-specific functionality or a reusable part of the public API. I opted for the latter.

It's an off-by-default optional feature, so it doesn't seem problematic to me for users who aren't interested in it.

Reply:
Def agree that it should be a part of the public API. I just don't think feature gating it is useful if we don't reduce the dependency tree by doing so.

I don't have very strong feelings about this, just a comment.

Reply:
`cosmos-sdk-proto` is being pulled in with `default-features = false`, so it isn't pulling in Tonic:

https://github.com/cosmos/cosmos-rust/blob/main/cosmos-tx/Cargo.toml#L14